tests/test_executor/test_executors.py

```diff
@@ -13,63 +13,93 @@
 
 
 class TestExecutor(TestEGCG):
+    executor_cls = Executor
+
+    def execute(self, *args, **kwargs):
+        e = self.executor_cls(*args, **kwargs)
+        try:
+            e.start()
+        except NotImplementedError:
+            pass
+
+        return e.join()
+
     def test_cmd(self):
-        e = Executor('ls ' + os.path.join(self.assets_path, '..'))
-        exit_status = e.join()
-        assert exit_status == 0
+        assert self.execute('ls ' + os.path.dirname(self.assets_path)) == 0
 
     def test_dodgy_cmd(self):
-        with pytest.raises(EGCGError) as err:
-            e = Executor('dodgy_cmd')
-            e.join()
-            assert 'Command failed: \'dodgy_cmd\'' in str(err)
+        assert self.execute('dodgy_cmd') == 127
```

Comment:
@mwhamgenomics What does the 127 mean? A comment would help future reviewers.

Reply:
127 comes from Bash attempting to run an unknown command:
```bash
mwhamgenomics@host $ nonexistent_cmd
bash: nonexistent_cmd: command not found
```
The aim of this test is to ensure that the exit status of Bash gets passed to the return value of `execute` - will comment/explain